src/services/accounts.js

```diff
@@ -0,0 +1,201 @@
+'use strict'
+
+const _ = require('lodash')
+const EventEmitter = require('eventemitter2')
+const compat = require('ilp-compat-plugin')
+const Store = require('../services/store.js')
+const Config = require('./config')
+const logger = require('../common/log')
+const log = logger.create('accounts')
+const { createIlpRejection, codes } = require('../lib/ilp-errors')
+
+const PLUGIN_EVENTS = [
+  'connect',
+  'disconnect',
+  'incoming_transfer',
+  'incoming_prepare',
+  'outgoing_fulfill',
+  'outgoing_cancel',
+  'outgoing_reject'
+]
+
+class Accounts extends EventEmitter {
+  constructor (deps) {
+    super()
+
+    const config = deps(Config)
+    this.store = deps(Store)
+
+    this._pluginList = [] // LedgerPlugin[]
+    this.plugins = {} // { prefix ⇒ LedgerPlugin }
+    this._prefixReverseMap = new Map() // { LedgerPlugin ⇒ prefix }
+    this._config = config
+    this._accounts = new Map()
+    this._accountInfo = new Map()
+    this.createIlpRejection = createIlpRejection.bind(null, config.address)
+    this._dataHandler = null
+    this._moneyHandler = null
+
+    const accounts = this
+    this._relayEvents = {}
+    PLUGIN_EVENTS.forEach((event) => {
+      this._relayEvents[event] = function () {
+        const args = Array.prototype.slice.call(arguments)
+        return accounts.emitAsync.apply(accounts, [event, this].concat(args))
+      }
+    })
+  }
+
+  connect (options) {
+    return Promise.all(this._pluginList.map((plugin) => plugin.connect(options)))
+  }
+
+  getPlugin (accountAddress) {
+    if (accountAddress.slice(-1) === '.') {
+      throw new Error('peer address must not end with "."')
+    }
+    return this.plugins[accountAddress] || null
+  }
+
+  getPlugins () {
+    return this._pluginList
+  }
+
+  getPrefixes () {
+    return Object.keys(this.plugins)
+  }
+
+  getCurrency (accountAddress) {
+    const account = this._accounts.get(accountAddress)
+
+    if (!account) {
+      log.debug('no currency found. account=' + accountAddress)
+      return null
+    }
+
+    return account.currency
+  }
+
+  add (accountAddress, creds, tradesTo, tradesFrom) {
+    log.info('add account. peerAddress=' + accountAddress)
+
+    creds = _.cloneDeep(creds)
+
+    if (!this._config.validateAccount(accountAddress, creds)) {
+      throw new Error('error while adding account, see error log for details.')
+    }
+
+    const store = this.store.getPluginStore(accountAddress)
+
+    creds.options.prefix = accountAddress
```

Comment:
Is this line necessary (see line 93 `prefix: accountAddress + '.',`)? If it is, shouldn't the prefix be set with a ` + '.'` suffix?

Reply:
Removed